src/test/java/io/leangen/graphql/ObjectScalarTest.java

```diff
@@ -0,0 +1,36 @@
+package io.leangen.graphql;
+
+import graphql.ExecutionResult;
+import graphql.GraphQL;
+import graphql.execution.batched.BatchedExecutionStrategy;
+import graphql.schema.GraphQLSchema;
+import io.leangen.graphql.annotations.GraphQLArgument;
+import io.leangen.graphql.annotations.GraphQLQuery;
+import io.leangen.graphql.domain.JsonPatch;
+import org.junit.Test;
+
+import java.util.List;
+
+import static org.junit.Assert.assertTrue;
+
+public class ObjectScalarTest {
+
+    public static class PatchService {
+
+        @GraphQLQuery(name = "processPatches")
+        public List<JsonPatch> processPatches(@GraphQLArgument(name = "args") List<JsonPatch> patches) {
+            return patches;
+        }
+    }
+
+    @Test
+    public void testObjectScalar() {
+        GraphQLSchema schema = new TestSchemaGenerator()
+                .withOperationsFromSingleton(new PatchService())
+                .generate();
+
+        GraphQL exe = GraphQL.newGraphQL(schema).queryExecutionStrategy(new BatchedExecutionStrategy()).build();
+        ExecutionResult result = exe.execute("{processPatches(args: {op: replace, path: \"/active\", value: 1}){op}}");
```

Comment:
The field is mapped as an object scalar, but this line only sends an integer... Why not map the field as an Int scalar?
Same question as above, is it allowed to be both?

Reply:
It is not only Int, it can be String,Boolean or even a Map

Let me explain, what I am trying to do, I want to implement JSON Patch (http://jsonpatch.com/) in a GraphQL mutation. Is there any other way to achieve this in GraphQL ?
My be i got the wrong idea about GraphQLScalar.

Reply:
Nope, you got the right idea. If it can be of multiple types, that is really the only way to go. And SPQR's object scalar should indeed be able to deal with simple values in addition to objects, for cases like this.